memory/src/main/scala/filodb.memory/format/vectors/HistogramVector.scala

```diff
@@ -468,7 +473,7 @@ class RowHistogramReader(histVect: Ptr.U8) extends HistogramReader with SectionR
     require(length > 0 && start >= 0 && end < length)
     val summedHist = MutableHistogram.empty(buckets)
     for { i <- start to end optimized } {
-      summedHist.addNoCorrection(apply(i).asInstanceOf[HistogramWithBuckets])
+      summedHist.addNoCorrection(apply(i))
```

Comment:
Why is the above not needed anymore?

Reply:
The type cast was redundant and not necessary originally. apply already returned right type